keanu-project/src/main/java/io/improbable/keanu/vertices/ProxyVertex.java

```diff
@@ -0,0 +1,11 @@
+package io.improbable.keanu.vertices;
+
+/**
+ * A Proxy vertex is one which has no parents at creation time but can at a later point "hook up" a parent node.  All
+ * Vertex operations are then passed through to the parent (as if the they'd been called on the parent directly)
```

Comment:
This description is true of your classes that implement `ProxyVertex`, but it's not a true description of the `ProxyVertex` interface itself. I think something along these lines is more accurate:

     * A Proxy vertex is one with a single parent. It is expected that all vertex operations will be delegated to the parent.
     * It typically has no parents at creation time - the parent is node is "hooked up" at a later point